src/config.rs

```diff
@@ -92,20 +104,27 @@ impl PbftConfig {
 pub fn load_pbft_config(block_id: BlockId, service: &mut Service) -> PbftConfig {
     let mut config = PbftConfig::default();
 
-    let settings: HashMap<String, String> = service
-        .get_settings(
-            block_id,
-            vec![
-                String::from("sawtooth.consensus.pbft.peers"),
-                String::from("sawtooth.consensus.pbft.block_duration"),
-                String::from("sawtooth.consensus.pbft.faulty_primary_timeout"),
-                String::from("sawtooth.consensus.pbft.view_change_duration"),
-                String::from("sawtooth.consensus.pbft.forced_view_change_period"),
-                String::from("sawtooth.consensus.pbft.message_timeout"),
-                String::from("sawtooth.consensus.pbft.max_log_size"),
-            ],
-        )
-        .unwrap_or_else(|err| panic!("Failed to load on-chain settings due to error: {:?}", err));
+    debug!("Getting on-chain settings for config");
+    let settings: HashMap<String, String> = retry_until_ok(
+        config.exponential_retry_base,
+        config.exponential_retry_max,
+        || {
+            service.get_settings(
+                block_id.clone(),
+                vec![
+                    String::from("sawtooth.consensus.pbft.peers"),
+                    String::from("sawtooth.consensus.pbft.block_duration"),
+                    String::from("sawtooth.consensus.pbft.exponential_retry_base"),
```

Comment:
Isn't this a retry that is based around the local configuration? I.e. it's a timeout for the messaging between the  consensus engine and the validator.